docs/reference/mapping/fields/synthetic-source.asciidoc

```diff
@@ -0,0 +1,170 @@
+[[synthetic-source]]
+==== Synthetic `_source`
+
+Though very handy to have around, the source field takes up a significant amount
+of space on disk. Instead of storing source documents on disk exactly as you
+send them, Elasticsearch can reconstruct source content on the fly. Enable this
+by setting `synthetic: true` in `_source`:
+
+[source,console,id=enable-synthetic-source-example]
+----
+PUT idx
+{
+  "mappings": {
+    "_source": {
+      "synthetic": true
```

Comment:
not a question for the docs specifically but users may wonder who takes the precedence between enabled and synthetic here if both are specified?

Reply:
`enabled: false, synthetic: true` is rejected. Or, at least, it will be. I'm sad to make something like that, but it's kind of required for backwards compatibility.

Reply:
why is that? folks that have disabled source in the past, could potentially switch to synthetic source if they have all fields with doc_values etc.?

Reply:
or especially if we were to synthetize from stored fields too?

Reply:
`synthetic` can't be modified either and old indices default to false. Mostly because I didn't want to have to think about what enabling it would mean. I hadn't really thought about folks with large old indices and source disabled. For what it's worth, there is a good chance it won't work either - it is the `picky` implementation of synthetic source after all.

Reply:
I see, maybe this could be something to consider and think about as a follow-up